src/api/schema/metrics.rs

```diff
@@ -0,0 +1,155 @@
+use crate::event::{Event, Metric, MetricValue};
+use crate::metrics::{capture_metrics, get_controller};
+use async_graphql::{validators::IntRange, Interface, Object, Subscription};
+use async_stream::stream;
+use chrono::{DateTime, Utc};
+use tokio::stream::{Stream, StreamExt};
+use tokio::time::Duration;
+
+pub struct Uptime(Metric);
+
+#[Object]
+impl Uptime {
+    /// Metric timestamp
+    async fn timestamp(&self) -> Option<DateTime<Utc>> {
+        self.0.timestamp
+    }
+
+    /// Number of seconds the Vector instance has been alive
+    async fn seconds(&self) -> f64 {
+        match self.0.value {
+            MetricValue::Gauge { value } => value,
+            _ => 0.00,
+        }
+    }
+}
+
+impl From<Metric> for Uptime {
+    fn from(m: Metric) -> Self {
+        Self(m)
+    }
+}
+
+pub struct EventsProcessed(Metric);
+
+#[Object]
+impl EventsProcessed {
+    /// Metric timestamp
+    async fn timestamp(&self) -> Option<DateTime<Utc>> {
+        self.0.timestamp
+    }
+
+    /// Number of events processed
+    async fn events_processed(&self) -> f64 {
+        match self.0.value {
+            MetricValue::Counter { value } => value,
+            _ => 0.00,
+        }
+    }
+}
+
+impl From<Metric> for EventsProcessed {
+    fn from(m: Metric) -> Self {
+        Self(m)
+    }
+}
+
+pub struct BytesProcessed(Metric);
+
+#[Object]
+impl BytesProcessed {
+    /// Metric timestamp
+    async fn timestamp(&self) -> Option<DateTime<Utc>> {
+        self.0.timestamp
+    }
+
+    /// Number of bytes processed
+    async fn bytes_processed(&self) -> f64 {
+        match self.0.value {
+            MetricValue::Counter { value } => value,
+            _ => 0.00,
+        }
+    }
+}
+
+impl From<Metric> for BytesProcessed {
+    fn from(m: Metric) -> Self {
+        Self(m)
+    }
+}
+
+#[Interface(field(name = "timestamp", type = "Option<DateTime<Utc>>"))]
+pub enum MetricType {
+    Uptime(Uptime),
+    EventsProcessed(EventsProcessed),
+    BytesProcessed(BytesProcessed),
+}
+
+#[derive(Default)]
+pub struct MetricsSubscription;
+
+#[Subscription]
+impl MetricsSubscription {
+    /// Metrics for how long the Vector instance has been running
+    async fn uptime_metrics(
+        &self,
+        #[arg(default = 1000, validator(IntRange(min = "100", max = "60_000")))] interval: i32,
```

Comment:
Looks like the `default` arg was fixed, and this now gets encoded as `Int` (optional) vs `Int!` (required)